### ISSUE TITLE
Cron Job failing

### ISSUE THREAD

**Issue by gbanina** [freescout-helpdesk/freescout#2184]
Hi.

I have an issue with the cron jobs on last version of FreeScout with the Workflows module active.

I keep getting this error:

```
Unknown named parameter $id {"exception":"[object] (Symfony\\Component\\Debug\\Exception\\FatalThrowableError(code: 0): Unknown named parameter $id at /var/www/html/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php:221)
[stacktrace]
#0 [internal function]: Illuminate\\Support\\Facades\\Facade::__callStatic()
#1 /var/www/html/app/Jobs/TriggerAction.php(42): call_user_func_array()
#2 [internal function]: App\\Jobs\\TriggerAction->handle()
#3 /var/www/html/overrides/laravel/framework/src/Illuminate/Container/BoundMethod.php(29): call_user_func_array()
#4 /var/www/html/overrides/laravel/framework/src/Illuminate/Container/BoundMethod.php(87): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#5 /var/www/html/overrides/laravel/framework/src/Illuminate/Container/BoundMethod.php(31): Illuminate\\Container\\BoundMethod::callBoundMethod()
#6 /var/www/html/overrides/laravel/framework/src/Illuminate/Container/Container.php(549): Illuminate\\Container\\BoundMethod::call()
#7 /var/www/html/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(94): Illuminate\\Container\\Container->call()
#8 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(114): Illuminate\\Bus\\Dispatcher->Illuminate\\Bus\\{closure}()
#9 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(102): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#10 /var/www/html/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(98): Illuminate\\Pipeline\\Pipeline->then()
#11 /var/www/html/vendor/laravel/framework/src/Illuminate/Queue/CallQueuedHandler.php(49): Illuminate\\Bus\\Dispatcher->dispatchNow()
#12 /var/www/html/vendor/laravel/framework/src/Illuminate/Queue/Jobs/Job.php(76): Illuminate\\Queue\\CallQueuedHandler->call()
#13 /var/www/html/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(320): Illuminate\\Queue\\Jobs\\Job->fire()
#14 /var/www/html/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(270): Illuminate\\Queue\\Worker->process()
#15 /var/www/html/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(114): Illuminate\\Queue\\Worker->runJob()
#16 /var/www/html/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(101): Illuminate\\Queue\\Worker->daemon()
#17 /var/www/html/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(85): Illuminate\\Queue\\Console\\WorkCommand->runWorker()
#18 [internal function]: Illuminate\\Queue\\Console\\WorkCommand->handle()
#19 /var/www/html/overrides/laravel/framework/src/Illuminate/Container/BoundMethod.php(29): call_user_func_array()
#20 /var/www/html/overrides/laravel/framework/src/Illuminate/Container/BoundMethod.php(87): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#21 /var/www/html/overrides/laravel/framework/src/Illuminate/Container/BoundMethod.php(31): Illuminate\\Container\\BoundMethod::callBoundMethod()
#22 /var/www/html/overrides/laravel/framework/src/Illuminate/Container/Container.php(549): Illuminate\\Container\\BoundMethod::call()
#23 /var/www/html/vendor/laravel/framework/src/Illuminate/Console/Command.php(183): Illuminate\\Container\\Container->call()
#24 /var/www/html/vendor/symfony/console/Command/Command.php(255): Illuminate\\Console\\Command->execute()
#25 /var/www/html/vendor/laravel/framework/src/Illuminate/Console/Command.php(170): Symfony\\Component\\Console\\Command\\Command->run()
#26 /var/www/html/vendor/symfony/console/Application.php(992): Illuminate\\Console\\Command->run()
#27 /var/www/html/vendor/symfony/console/Application.php(255): Symfony\\Component\\Console\\Application->doRunCommand()
#28 /var/www/html/vendor/symfony/console/Application.php(148): Symfony\\Component\\Console\\Application->doRun()
#29 /var/www/html/vendor/laravel/framework/src/Illuminate/Console/Application.php(88): Symfony\\Component\\Console\\Application->run()
#30 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(121): Illuminate\\Console\\Application->run()
#31 /var/www/html/artisan(62): Illuminate\\Foundation\\Console\\Kernel->handle()
#32 {main}
"}
```

this is an example of a job that is failing:

`{"displayName":"App\\Jobs\\TriggerAction","job":"Illuminate\\Queue\\CallQueuedHandler@call","maxTries":null,"timeout":null,"timeoutAt":null,"data":{"commandName":"App\\Jobs\\TriggerAction","command":"O:22:\"App\\Jobs\\TriggerAction\":9:{s:6:\"action\";s:19:\"workflow.do_process\";s:6:\"params\";a:1:{s:2:\"id\";i:4;}s:6:\"\u0000*\u0000job\";N;s:10:\"connection\";N;s:5:\"queue\";s:7:\"default\";s:15:\"chainConnection\";N;s:10:\"chainQueue\";N;s:5:\"delay\";N;s:7:\"chained\";a:0:{}}"}}`

The serialized part doesn't look good ...

App Version | 1.8.21
Web Server | nginx/1.18.0
PHP Version | PHP 8.0.19

please advise.
tnx.

**Comment by freescout-helpdesk** [comment]
It's hard to tell what can be the reason. Are you using any non-official modules?

**Comment by gbanina** [comment]
> It's hard to tell what can be the reason. Are you using any non-official modules?

Hi. Thank you for your reply. 
No. All my modules are from https://freescout.net/modules/ 

The list of my active modules:

API and Webhooks
Custom Fields
Custom Folders
Customization & RebrandingActive
Knowledge Base
Reports
Saved Replies
Tags
Ticket Translator
Time Tracking
Widgets White-Labeling
Workflows

It looks like the issue occurs when one of my (multiple) Workflows triggers, but looking at the stack trace, I don't really see what condition would trigger it. 

Do you have any idea what steps to take to continue troubleshooting this?

**Comment by freescout-helpdesk** [comment]
Add on line 41 in https://github.com/freescout-helpdesk/freescout/blob/dist/app/Jobs/TriggerAction.php#L41 the following and clear the cache:

```
\Log::error('TriggerAction args: '.json_encode($args));
```
After that you will see see 'TriggerAction args' in App Logs - post it here.

**Comment by gbanina** [comment]
> Add on line 41 in https://github.com/freescout-helpdesk/freescout/blob/dist/app/Jobs/TriggerAction.php#L41 the following and clear the cache:
> 
> ```
> \Log::error('TriggerAction args: '.json_encode($args));
> ```
> 
> After that you will see see 'TriggerAction args' in App Logs - post it here.

Yes, I have already tried this:

\Log::error('Chron log -' . $action . '- ' . json_encode($args));

Log is :

Chron log -workflow.do_process- {"0":"workflow.do_process","id":4}

**Comment by freescout-helpdesk** [comment]
Fixed in Workflows Module v1.0.27
